lib/rouge/lexers/apex.rb

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class Apex < RegexLexer
+      title "Apex"
+      desc "The Apex programming language (provided by salesforce)"
+
+      tag 'apex'
+      filenames '*.cls'
+      mimetypes 'text/x-apex'
+
+      keywords = %w(
```

Comment:
To reduce Rouge's base load memory usage, it's best to wrap your local variables (like `keywords`, `declarations`, `soql`, etc) in a class method. This isn't something that all lexers do yet but we're trying to enforce this for new lexers as we update the old ones.

So rather than:

```rb
keywords = %w(
  assert break case catch continue default do else finally for
  if goto instanceof new return switch this throw try while insert
  update delete
)
```

you'd write:

```rb
def self.keywords
  @keywords ||= %w(
    assert break case catch continue default do else finally for
    if goto instanceof new return switch this throw try while insert
    update delete
  )
end
```